projects/components/src/dropdown/dropdown-focus-handler.directive.ts

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { AfterViewInit, Directive, Host, OnDestroy, Optional, SkipSelf } from '@angular/core';
+import { ClrPopoverToggleService } from '@clr/angular';
+import { Subscription } from 'rxjs';
+import { SubscriptionTracker } from '../common/subscription';
+import { DropdownFocusHandlerService } from './dropdown-focus-handler.service';
+import { DropdownComponent } from './dropdown.component';
+
+const BUTTON_NODE_NAME = 'BUTTON';
+
+/**
+ * Arrow keys directions
+ */
+export enum Direction {
+    UP = 'up',
+    DOWN = 'down',
+    LEFT = 'left',
+    RIGHT = 'right',
+}
+
+/**
+ * Object wrapping the focusable HTML elements of dropdown menu with the neighbors in 4 directions of each menu item.
+ */
+export interface MenuItem {
+    element: HTMLElement;
+    up?: MenuItem;
+    down?: MenuItem;
+    left?: MenuItem;
+    right?: MenuItem;
+    closeMenu?: (event: Event) => void;
+}
+
+/**
+ * Added on to vcd-dropdown component to vertically link menu items of the dropdown on which this directive is added. It also, then links
+ * the vertically linked menu items horizontally to their menu trigger. It then uses the {@link DropdownFocusHandlerService} to move the
+ * DOM focus between the menu items.
+ *
+ * @Example:
+ * <vcd-dropdown
+ *        vcdDropdownFocusHandler
+ *        [items]="contextualActions"
+ *        [onItemClickedCb]="runActionHandler.bind(this)"
+ *        [isItemDisabledCb]="isActionDisabled.bind(this)"
+ * ></vcd-dropdown>
+ */
+@Directive({
+    selector: 'vcd-dropdown[vcdDropdownFocusHandler]',
+})
+export class DropdownFocusHandlerDirective<T> implements AfterViewInit, OnDestroy {
+    constructor(
+        @Optional() @SkipSelf() private parentVcdDropdown: DropdownComponent<T>,
+        @Optional() @SkipSelf() private parentFocusHandler: DropdownFocusHandlerDirective<T>,
+        @Host() private hostVcdDropdown: DropdownComponent<T>,
+        private focusHandlerService: DropdownFocusHandlerService
+    ) {}
+
+    /**
+     * List of focusable menu items with their neighbors in 4 directions.
+     */
+    menuItems: MenuItem[];
+
+    private menuTrigger: MenuItem;
```

Comment:
Can we either put a comment on this property or rename it to be more clear? I'm not totally sure what it represents

Reply:
This is just a trigger that opened this menu